### PR TITLE
allow overwriting existing registry entries

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request adds support for overwriting existing registry entries, which is particularly useful for downstream apps that want to modify something already in the registry without having to give it a new name.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/registries.py [hunk 1]
```diff
@@ -75,10 +77,12 @@ def add(self, name, cls, label=None):
             in the first parameter.
         label : str, optional
             The label displayed in the tooltip when hovering over the tray tab.
+        overwrite : bool, optional
+            Whether to overwrite an existing entry with the same ``label``.
```
Comment (pllim) [inline jdaviz/core/registries.py:81]: Maybe add one more sentence to explain what would happen to the old thing that was overwritten?

Comment (kecnry) [reply jdaviz/core/registries.py:81]: suggestions for wording?  It is replacing it in the registry, so calls to access the item will retrieve the new one instead.

Comment (pllim) [reply jdaviz/core/registries.py:81]: Is the old one going to be destroyed somewhere or just sitting in purgatory? I am not familiar with this machinery.

Comment (kecnry) [reply jdaviz/core/registries.py:81]: This doesn't do anything to destroy it, no, it just replaces the reference in the registry.

Comment (pllim) [reply jdaviz/core/registries.py:81]: Should we destroy it?

Comment (kecnry) [reply jdaviz/core/registries.py:81]: I don't see any reason to, its generally just a class and shouldn't be instantiated yet

**Review by pllim — APPROVED** [review]
Code looks like code. Thanks!

**Comment by pllim** [comment]
It is Friday and the weather is nice, let's get this in. 😆

**Pull request merged**